cnf-certification-test/platform/hugepages/hugepages_test.go

```diff
@@ -1 +1,716 @@
+	testKernelArgsHpDefSizePlusPairs3 = []string{"systemd.cpu_affinity=0,1,40,41,20,21,60,61", "default_hugepagesz=1G", "hugepagesz=1G", "hugepages=16", "hugepagesz=2M", "hugepages=256", "nmi_watchdog=0"}
+)
+
+const (
+	// Sizes, in KBs.
+	oneGB = 1024 * 1024 // 1G
+	twoMB = 2 * 1024    // 2M: also RHEL's default hugepages size
+)
+
+//nolint:funlen
+func Test_hugepagesFromKernelArgsFunc(t *testing.T) {
+	testCases := []struct {
+		expectedHugepagesDefSize int
+		expectedHugepagesPerSize map[int]int
+		kernelArgs               []string
+	}{
+		// No params
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{twoMB: 0},
+			kernelArgs:               testKernelArgsHpNoParams,
+		},
+
+		// Single params TCs.
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{twoMB: 16},
+			kernelArgs:               testKernelArgsHpSingleParam1,
+		},
+		{
+			expectedHugepagesDefSize: oneGB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 0},
+			kernelArgs:               testKernelArgsHpSingleParam2,
+		},
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{twoMB: 0},
+			kernelArgs:               testKernelArgsHpSingleParam3,
+		},
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 0},
+			kernelArgs:               testKernelArgsHpSingleParam4,
+		},
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 16},
+			kernelArgs:               testKernelArgsHpPair1,
+		},
+
+		// Default sizes Tc:
+		{
+			expectedHugepagesDefSize: oneGB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 0},
+			kernelArgs:               testKernelArgsHpDefParamsOnly,
+		},
+
+		// size+count pairs
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 16},
+			kernelArgs:               testKernelArgsHpPair1,
+		},
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{twoMB: 256},
+			kernelArgs:               testKernelArgsHpPair2,
+		},
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 16, twoMB: 256},
+			kernelArgs:               testKernelArgsHpPair3,
+		},
+
+		// default size + (size+count) pairs
+		{
+			expectedHugepagesDefSize: twoMB,
+			expectedHugepagesPerSize: map[int]int{twoMB: 0, oneGB: 16},
+			kernelArgs:               testKernelArgsHpDefSizePlusPairs1,
+		},
+		{
+			expectedHugepagesDefSize: oneGB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 0, twoMB: 256},
+			kernelArgs:               testKernelArgsHpDefSizePlusPairs2,
+		},
+		{
+			expectedHugepagesDefSize: oneGB,
+			expectedHugepagesPerSize: map[int]int{oneGB: 16, twoMB: 256},
+			kernelArgs:               testKernelArgsHpDefSizePlusPairs3,
+		},
+	}
+
+	mc := provider.MachineConfig{MachineConfig: &mcv1.MachineConfig{}}
```

Comment:
Why not create the `mc` object every iteration?

Reply:
Done